File: largestofFour.js
// Sort an array of arrays
// Separate the biggest number in each 
// array to a new array

function largstofFour(arr) {
    let output = [];
    for (var i = 0; i < arr.length; i++){
        arr[i].sort((a, b) => a - b);
        output.push(arr[i][arr[i].length - 1]);
    }
    console.log(output);
}

function largstofFour2(arr) {
    let output = [];
    for (var i = 0; i < arr.length; i++){
        let largestEach = -10000000;
        for (let j = 0; j < arr[i].length; j++){
            if (arr[i][j] > largestEach){
                largestEach = arr[i][j];
            }
        }
        output.push(largestEach);
    }
    console.log(output);
    return (output);
}

largstofFour2([[17, 23, 25, 12], 
    [25, 7, 34, 48], 
    [4, -10, 18, 21], 
    [-72, -3, -17, -10]]);

File: largestofFour.rb
# Sort an array of arrays
# Separate the biggest number in each
# array to a new array
def largestofFour(arr)
  arr.map { |e| e.sort.last }.sort
end

def largestofFour2(arr)
  output = []
  arr.each do |array|
    output << array.sort.last
  end
  output.sort
end


p largestofFour2([[4, 5, 1, 3, 8, 50], [13, 27, 18, 26], [32, 35, 37, 39], [1000, 1001, 857, 1]])
